src/main/java/mate/academy/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,40 @@
+package mate.academy.security;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDbOptional = userService.findByEmail(email);
+        if (userFromDbOptional.isEmpty() || !passwordCheck(userFromDbOptional, password)) {
+            throw new AuthenticationException("Can't authenticate user with email: " + email);

```

Comment:
краще не вказувати по якій причині саме ми не можемо аутентифікуватись
```suggestion
            throw new AuthenticationException("Can't authenticate user, email or password is incorrect);

```